### PR TITLE
Support for ping check to get healthy connections

### LINKED ISSUE #8: MySQL Server has gone away
The pooled connection might to be idle for a long time, and the server may disconnect or kill the process for any reason. In this case, the broker should check the connection before deliver it to the requester. It would be nice if PyMySQLPool do a `ping()` after some amount of time to force reconnection.

### PR DESCRIPTION
**Pull request by kawamanza**
Closes #8

### CODE REVIEW

File: pymysqlpool/pool.py [hunk 1]
```diff
@@ -141,23 +145,43 @@ def _wait(self):
...
+                timeout = timeout - self.ping_check
+            if not hasattr(c, '__ping_check_timestamp'):
+                c.__ping_check_timestamp = now
+            try:
+                if c.__ping_check_timestamp < timeout:
+                    c.__ping_check_timestamp = now
+                    c.ping()
+            except:
+                self.current_size -= 1
+                if retry_count < 10: c = self.__get_conn(retry_count+1)
```
Comment (prprprus) [inline pymysqlpool/pool.py:182]: Assume retry exceeds 10, `ping()` still fails. Is there need to remove this invalid conn and recreate a conn, or throw an exception?

Comment (kawamanza) [reply pymysqlpool/pool.py:182]: When the `ping()` fails, the _conn_ is discarded and another _conn_ is fetched from the pool. The _retry_count_ is used to to get and return the first healthy _conn_ from the pool a couple of times. If the _conn_ object fails on trying reconnect, maybe the object has some problem with its internal properties. But... if other instances have the same problem, maybe there is a network problem on going, and no _conn_ will be returned anymore. This implementation just return a `None` _conn_ on a network problem occurring, but we can make a custom exception if you prefer.

**Comment by prprprus** [comment]
Oh, yes, `ping()` will [reconnect](https://pymysql.readthedocs.io/en/latest/modules/connections.html?highlight=ping#pymysql.connections.Connection.ping), I forgot it 😅...

Nice pr 👍🎉👏, merged.

File: pymysqlpool/pool.py (hunk 1 above)
Comment (prprprus) [reply pymysqlpool/pool.py:182]: Oh, `ping()` will [reconnect](https://pymysql.readthedocs.io/en/latest/modules/connections.html?highlight=ping#pymysql.connections.Connection.ping), I forgot it 😅...

Nice pr 👍🎉👏, merged.

**Pull request merged**